database/RocksProperties.java

```diff
@@ -215,43 +275,35 @@ public void run() {
-                        builder.append(format("%-40s %s\n", property.label(), property.getFormatted(rocksDB, cfHandles)));
-                    } else {
-                        builder.append(format("%-40s %s\n", property.label(), property.getFormatted(rocksDB)));
-                    }
-                }
-                LOG.debug(builder.toString());
-            } catch (RocksDBException e) {
-                LOG.error(STORAGE_PROPERTY_EXCEPTION.message(), e);
+            StringBuilder builder = new StringBuilder(format(
+                    "Database '%s' rocksdb properties from '%d' column families:\n", database, reader.cfHandles.size()));
+            for (Property property : RocksProperties.properties.values()) {
+                builder.append(format("%-40s %s\n", property.label(), reader.getFormatted(property)));
             }
+            LOG.debug(builder.toString());
         }
 
         private void logRocksFilesSummary() {
             try {
-                long[] dataBlocksSize = new long[cfHandles.size()];
+                long[] dataBlocksSize = new long[reader.cfHandles.size()];
                 Arrays.fill(dataBlocksSize, 0);
-                long[] indexBlocksSize = new long[cfHandles.size()];
+                long[] indexBlocksSize = new long[reader.cfHandles.size()];
                 Arrays.fill(indexBlocksSize, 0);
-                long[] filterBlocksSize = new long[cfHandles.size()];
+                long[] filterBlocksSize = new long[reader.cfHandles.size()];
                 Arrays.fill(filterBlocksSize, 0);
-                long[] rawKeysSize = new long[cfHandles.size()];
+                long[] rawKeysSize = new long[reader.cfHandles.size()];
                 Arrays.fill(rawKeysSize, 0);
-                long[] rawValuesSize = new long[cfHandles.size()];
+                long[] rawValuesSize = new long[reader.cfHandles.size()];
                 Arrays.fill(rawValuesSize, 0);
-                long[] numEntries = new long[cfHandles.size()];
+                long[] numEntries = new long[reader.cfHandles.size()];
                 Arrays.fill(numEntries, 0);
-                long[] numDeletions = new long[cfHandles.size()];
+                long[] numDeletions = new long[reader.cfHandles.size()];
                 Arrays.fill(numDeletions, 0);
-                long[] keysEstimate = new long[cfHandles.size()];
+                long[] keysEstimate = new long[reader.cfHandles.size()];
                 long sstFiles = 0;
-                for (int i = 0; i < cfHandles.size(); i++) {
-                    ColumnFamilyHandle cfHandle = cfHandles.get(i);
-                    Map<String, TableProperties> sstProperties = rocksDB.getPropertiesOfAllTables(cfHandle);
+                for (int i = 0; i < reader.cfHandles.size(); i++) {
+                    ColumnFamilyHandle cfHandle = reader.cfHandles.get(i);
+                    Map<String, TableProperties> sstProperties = reader.rocksDB.getPropertiesOfAllTables(cfHandle);
```

Comment:
Let's test that these measurements are meaningful over time.

Reply:
Checked :)